src/AspNet.Security.OAuth.Shopify/ShopifyAuthenticationHandler.cs

```diff
@@ -0,0 +1,218 @@
+                
+                var userData = tokens.Response["associated_user"].ToString();
+                identity.AddClaim(new Claim(ClaimTypes.UserData, userData));
+            }
+
+            identity.AddClaim(new Claim(ClaimTypes.IsPersistent, isPersistent.ToString(), ShopifyAuthenticationOptions.XmlSchemaBoolean));
+            identity.AddClaim(new Claim(ShopifyAuthenticationDefaults.ShopifyScopeClaimType, actualScope));
+
+            var principal = new ClaimsPrincipal(identity);
+
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, payload);
+            context.RunClaimActions(payload);
+
+            await Options.Events.CreatingTicket(context);
+            var ticket = new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
+
+            return ticket;
+        }
+
+        /// <inheritdoc />
+        protected override string FormatScope()
+        {
+            return string.Join(",", Options.Scope);
+        }
+
+        /// <inheritdoc />
+        protected override string BuildChallengeUrl(AuthenticationProperties properties, string redirectUri)
+        {
+            if (!properties.Items.ContainsKey(ShopifyAuthenticationDefaults.ShopNameAuthenticationProperty))
+            {
+                var msg =
+                    $"Shopify provider AuthenticationProperties must contain {ShopifyAuthenticationDefaults.ShopNameAuthenticationProperty}";
+
+                Logger.LogError(msg);
+                throw new Exception(msg);
+            }
+
+            var shopName = properties.Items[ShopifyAuthenticationDefaults.ShopNameAuthenticationProperty];
+            var uri = string.Format(Options.AuthorizationEndpoint, shopName);
+
+            // Get the permission scope, which can either be set in options or overridden in AuthenticationProperties.
+            var scope = properties.Items.ContainsKey(ShopifyAuthenticationDefaults.ShopScopeAuthenticationProperty) ? 
+                properties.Items[ShopifyAuthenticationDefaults.ShopScopeAuthenticationProperty] : 
+                FormatScope();
+
+            var url = QueryHelpers.AddQueryString(uri, new Dictionary<string, string>
+            {
+                ["client_id"] = Options.ClientId,
+                ["scope"] = scope,
+                ["redirect_uri"] = redirectUri,
+                ["state"] = Options.StateDataFormat.Protect(properties)
+            });
+
+            // If we're requesting a per-user, online only, token, add the grant_options query param.
+            if (properties.Items.ContainsKey(ShopifyAuthenticationDefaults.GrantOptionsAuthenticationProperty))
+            {
+                var grantOptions = properties.Items[ShopifyAuthenticationDefaults.GrantOptionsAuthenticationProperty];
+                if (grantOptions != null &&
+                    grantOptions.Equals(ShopifyAuthenticationDefaults.PerUserAuthenticationPropertyValue))
```

Comment:
Assuming this is a `string`, use `string.Equals` and explicitly state whether it is case sensitive or not using `StringComparisonOptions.Ordinal...`, then the null check also becomes redundant.